.github/workflows/is-test-v2.yml

```diff
@@ -22,11 +22,17 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
 
+      - name: Get Date
+        id: get-date
+        run: |
+          echo "::set-output name=date::$(/bin/date -u "+%Y-%m")"
+        shell: bash
+
       - name: Go build cache
         uses: actions/cache@v2
         with:
           path: ~/.cache/go-build
-          key: ${{ runner.os }}-go-build-${{ hashFiles('**/*.go', '**/go.mod', '**/go.sum') }}
+          key: ${{ runner.os }}-go-build-${{ steps.get-date.outputs.date }}
```

Comment:
I think these changes are going to implement a different kind of behaviors than expected.

The old behavior was supposed to create files based on the source code and go dependency files and update them if they change, otherwise use the cached version when they are similar to thefiles used in the previous runs thus not downloading the same dependency packages again, instead using them from cache.

The new behavior seems to replace the content-based hash with a date which is the current date in month if I see it correctly which is always going to use the current year and month for caching purposes.

That means,
1. every month the cache will be invalidated and the dependencies will be downloaded again (smaller issue), and
2. every time the source code or the go dependency files change, the dependencies are **NOT** going to be updated, until a new month begins (because of the date based caching), thus the dependency changes will not take effect.

I think this is going to cause further issues down the line - unless I misunderstood something about the caching here - so I think we should reconsider our approach here based on what problem we are trying to solve.

Reply:
I agree, it was only one of my many attempts to fix the original problem which is clearly not solved yet so it is a work in progress 🙂